### DISCUSSION TITLE
Allow pause of filter automatic updates when network is metered

### DISCUSSION THREAD

**Discussion by DeepChirp** [uBlockOrigin/uBlock-issues#2854] [Ideas] [answered]
Allow pause of filter automatic updates when network is metered to avoid additional traffic costs.

**Comment by ghost** [comment]
Chromium experimental API https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData

**Reply by gorhill** [reply]
Couldn't this lead to cases of filter lists never being updated?

**Reply by ghost** [reply]
Probably will. To be clear - I'm against. I doubt ads and tracking will pause on metered/slow connections. I thought it's not possible to detect connection type (no API), but I was wrong.

**Reply by gorhill** [reply]
Yeah probably first step would be to make the case that auto-downloading filter lists is causing undue brandwidth usage. I am skeptical of this given most lists update every few days, and the encoding is gzip during transfert.

**Reply by gorhill** [reply, marked as answer]
Here are some numbers which cover the lists served by uBO itself, so this exclude _Online Malicious URL Blocklist_ and _Peter Lowe’s Ad and tracking server list_ (which I small relative to other main lists).

The figures are derived from jsDelivr stats:

- https://www.jsdelivr.com/package/gh/gorhill/uBlock?tab=stats
- https://www.jsdelivr.com/package/gh/uBlockOrigin/uAssetsCDN?tab=stats

Question is: how much bandwidth per user per month is attributable to uBO's auto-update of filter lists.

---

First: how many users of uBO out there?

Answer: [this](https://www.jsdelivr.com/package/gh/gorhill/uBlock?tab=stats) says that `assets.json` is currently downloaded 48,317,831 per month. Given that:

- `assets.json` is set to update every 13 days
- jsDelivr is just one of 4 CDNs being used

Thus: requests-per-month / (days in a month / 13-day update period) x 4 CDNs = 48,317,831 / (30 / 13) x 4 = 83,750,907 users.

---

Second: How much bandwidth per month is consumed by all those users?

Answer: [this](https://www.jsdelivr.com/package/gh/uBlockOrigin/uAssetsCDN?tab=stats) says that bandwidth consumed in the last month is 96 TB.

Thus: per-month bandwidth x 4 CDNs = 96 TB x 4 = 384 TB = 384,000,000,000,000 per month.

Answer to _"how much bandwidth per user per month is attributable to uBO's auto-update of filter lists"_:

384,000,000,000,000 / 83,750,907 = 4,585,024 per month = 4.6 MB per month = 153 KB per day on average.

---

Using the same method for `assets.json` (which is pulled from a different repo (so separate jsDelivr stats page), we get 472 bytes per day on average.

Thus let's say 153 KB + 472 bytes = 154 KB per day.

Probably need to add a few KB to account `urlhaus-1`/`plowe-0` -- these lists are rather small compared to the rest, so I will assume the above figure is close enough to reality.

Of course, those who enabled more filter lists will require more bandwidth, the above numbers give a ballpark idea of bandwidth usage by uBO for default lists.

---

Conclusion: At an average of 154 KB per day, I don't see the need of a auto-pause auto-update feature as being of interest at this point, especially given that the bad side-effect is some installations potentially running with seriously out of date lists, and given this would just require to add more code which I rather not create/maintain unless really needed.

**Reply by krystian3w** [reply]
Maybe possible is force brotli instead gzip for compatible servers?

https://caniuse.com/brotli:

Firefox 44+ (legacy XP, Vista too)
Chrome 51+
Edge 79+
Opera 38+

**Reply by ghost** [reply]
Brotli is already used.

***

br:
cdn.statically.io
ublockorigin.pages.dev
cdn.jsdelivr.net
easylist.to
secure.fanboy.co.nz

gzip:
github.io
filters.adtidy.org
raw.githubusercontent.com
get.letsblock.it

plain text
pgl.yoyo.org
